src/main/java/hsien/Hsien.java

```diff
@@ -0,0 +1,260 @@
+        try {
+            FileWriter fw = new FileWriter("tasks.txt");
+            for (Task task: messages) {
+                fw.write(task.getStatusDescription() + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred while writing to the file.");
+        }
+    }
+
+    public static String[] processCommand(String input) {
+        String[] parts = input.split(" ");
+        String desc = String.join(" ", Arrays.copyOfRange(parts, 1, parts.length));
+        String command = parts[0];
+        return new String[]{command, desc};
+    }
+
+    public static String[] processFileLine(String input) throws HsienException{
+        String[] parts = input.split(" ");
+        System.out.println(input);
+        String regex = "\\[(.*?)\\] \\[(.*?)\\] (.*)";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(input.trim());
+
+        if (matcher.matches()) {
+            String command = matcher.group(1); // Extract command from the first group
+            String isMark = matcher.group(2); // Extract mark status from the second group
+            String desc = matcher.group(3); // Extract description from the third group
+            return new String[]{command, isMark, desc};
+        } else {
+            throw new HsienException();
+        }
+    }
+
+    public static void addTaskFromFile(String command, String isMark, String desc, ArrayList<Task> messages) {
+        Task newTask = null;
+        String tempDesc;
+
+        // Create Task object based on action
+        if (command.equals("T")) {
+            newTask = new Todo(desc);
+        } else if (command.equals("D")) {
+            tempDesc = desc.split("\\(by:")[0].trim();
+            String byDate = desc.split("\\(by:")[1].trim();
+            newTask = new Deadline(tempDesc, byDate);
+        } else {
+            tempDesc = desc.split("\\(from:")[0].trim();
+            String[] dates = desc.split("\\(from:")[1].split("to:");
+            String fromDate = dates[0].trim();
+            String toDate = dates[1].substring(0, dates[1].length()-1).trim();
+            newTask = new Event(tempDesc, fromDate, toDate);
+        }
+
+        // Mark task
+        if (isMark.equals("X")) {
+            newTask.mark();
+        }
+
+        messages.add(newTask);
+        System.out.println("Added task: " + newTask.getDescription());
+        System.out.printf("Now you have [%d] tasks in the list.%n", messages.size());
+    }
+
+    public static void addTask(String command, String desc, ArrayList<Task> messages) {
+        // Empty task
+        try {
+            if (desc.isEmpty()) {
+                throw new HsienException();
+            }
+        } catch (HsienException e) {
+            System.out.println("Description cannot be left empty");
+            printLine();
+            return;
+        }
+
+        Task newTask = null;
+        String tempDesc;
```

Comment:
Ensure names of variables are self explanatory. Consider 'description' or simpler names